django/db/backends/oracle/base.py

```diff
@@ -475,23 +468,17 @@ def _fix_for_params(self, query, params, unify_by_values=False):
             params_dict = {param: ':arg%d' % i for i, param in enumerate(set(params))}
             args = [params_dict[param] for param in params]
             params = dict(zip(params_dict.values(), params_dict.keys()))
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         else:
             # Handle params as sequence
             args = [(':arg%d' % i) for i in range(len(params))]
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         return query, self._format_params(params)
```

Comment:
Looks okay. By the way, there's one failure on Python 2 maybe you can advise on or investigate:
```
======================================================================
ERROR: setUpClass (expressions_case.tests.CaseExpressionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/django/test/testcases.py", line 1040, in setUpClass
    cls.setUpTestData()
  File "/home/tim/code/django/tests/expressions_case/tests.py", line 27, in setUpTestData
    o = CaseTestModel.objects.create(integer=1, integer2=1, string='1')
  File "/home/tim/code/django/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/tim/code/django/django/db/models/query.py", line 392, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/tim/code/django/django/db/models/base.py", line 804, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/tim/code/django/django/db/models/base.py", line 834, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/home/tim/code/django/django/db/models/base.py", line 920, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/home/tim/code/django/django/db/models/base.py", line 959, in _do_insert
    using=using, raw=raw)
  File "/home/tim/code/django/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/tim/code/django/django/db/models/query.py", line 1032, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/tim/code/django/django/db/models/sql/compiler.py", line 1045, in execute_sql
    cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/tim/code/django/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/backends/oracle/base.py", line 485, in execute
    return self.cursor.execute(query, self._param_generator(params))
DatabaseError: ORA-12704: character set mismatch

query/params:
INSERT INTO "EXPRESSIONS_CASE_CASETESTMODEL" ("INTEGER", "INTEGER2", "STRING",
 "BIG_INTEGER", "BINARY", "BOOLEAN", "COMMA_SEPARATED_INTEGER",
 "DATE_FIELD", "DATE_TIME", "DECIMAL_FIELD", "DURATION", "EMAIL", "FILE_FIELD",
 "FILE_PATH", "FLOAT_FIELD", "IMAGE", "GENERIC_IP_ADDRESS", "NULL_BOOLEAN", 
"POSITIVE_INTEGER", "POSITIVE_SMALL_INTEGER", "SLUG", "SMALL_INTEGER", "TEXT",
 "TIME_FIELD", "URL", "UUID", "FK_ID") VALUES (:arg1, :arg1, :arg5, :arg4, :arg3,
:arg0, :arg3,  :arg4, :arg4, :arg4, :arg4, :arg3, :arg3, :arg3, :arg4, :arg3, :arg4,
:arg4, :arg4, :arg4, :arg3, :arg4,  :arg3, :arg4, :arg3, :arg4, :arg4)
 RETURNING "EXPRESSIONS_CASE_CASETESTMODEL"."ID" INTO :arg2
{u':arg3': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddc6d0>,
 u':arg2': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddce90>,
 u':arg1': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddc450>,
 u':arg0': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddc5d0>,
 u':arg5': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddc3d0>,
 u':arg4': <django.db.backends.oracle.base.OracleParam object at 0x7fa3a5ddc890>}
```